### ISSUE TITLE
Ability to display resident's transaction history

### ISSUE THREAD

**Issue by rakhimosip** [mosip/inji#188]
**Problem statement:** A resident should be able to use Inji and view their transaction history.

**Proposed solution:**

- The app should have the option to view the transaction history of a resident
- To access this option the resident will have to login (either using UIN+ OTP or using OIDC token)
- There should be an option to set the total number of records returned. For example -> a default search should return the latest 50 records
- There should be an option to filter records by a date range
- The resident should be able to scroll through the records and view the details of transactions.
- The option to view transactions can be given in the Profile section of the app

**Comment by walidkhouryNL** [comment]
Hello @rakhimosip 
I have a quick question regarding the transaction history list: 
What would be the information detail a user can find under a transaction entry? (Transaction number, date & time, etc..) Could you please provide us with a list of info? 
Thank you
cc: @kyanthony

**Comment by rakhimosip** [comment]
@walidkhouryNL 
We can display the following details
- Transaction ID (can be a number <20 digits)
- Description
- Date and time
- Status (Can be an icon too)

One should have the ability to filter based on a date range, request Type, and status

**Comment by kneckinator** [comment]
@rakhimosip for the request types, which are the user-displayable strings? I guess we should not use `AUTHENTICATION_REQUEST`, `SERVICE_REQUEST`, `DATA_UPDATE_REQUEST` in the filter drop-down and when displaying the list of transactions.

> There should be an option to set the total number of records returned. For example -> a default search should return the latest 50 records
Can we agree to not make this settable in the app and have the app always return the last 50 transactions? I guess the user can use the resident portal if they want to dive deeper into their transaction history.

> The resident should be able to scroll through the records and view the details of transactions.
What would the detailed transaction screen look like and what information would it contain that is not in the list?

For API access, you mention that either UIN+OTP or OIDC token would work, but the API spec on stoplight.io only details the OIDC auth token.

**Comment by rakhimosip** [comment]
@kneckinator You are right, we should map the request types to strings from locale files and use them.

We can limit it to return 50 transaction records. But limiting it to only 'last 50' might not be that useful to the resident given that we have given them a date range-based search too.
So by default, we can return the last 50 transaction records and when the resident uses search fields like date range or request type, we can return the top 50 records.

We have an API that returns details of the transaction record. For example, in the case of a DATA_UPDATE_REQUEST transaction, the details would list out what resident attributes were updated. 
For now, we can limit the functionality to display just the transaction details like 

- Transaction ID (can be a number <20 digits)
- Description
- Date and time
- Status (Can be an icon too)
Displaying details can be taken up in the next phase if required.

The preference is to go with token-based API. I will remove UIN+OTP from the requirement.

CC @kyanthony

**Comment by kneckinator** [comment]
@rakhimosip Sorry I forgot to include a question regarding the date range which was related to the 50 transaction records. I was thinking that maybe the usage of the app would be a compliment to the MOSIP resident portal and as such, we might not need to set a date range.

What do you think about this idea?

**Comment by rakhimosip** [comment]
@kneckinator even if the user were to use resident portal features, I feel this feature implementation on the app should be usable on its own.
